diff --git a/test_api.py b/test_api.py
index 8b98c11..15dd681 100644
--- a/test_api.py
+++ b/test_api.py
@@ -13,19 +13,14 @@ def client():
     return app.test_client()
 
 
-def test_successful_api_call(client):
-    response = client.get('/?a=1&b=2')
-    assert response.status_code == 200
-
-
-def test_mandatory_params(client):
-    response = client.get('/?a=wrong&b=parameters')
-    assert response.status_code == 400
-
-
-def test_missing_params(client):
-    response = client.get('/')
-    assert response.status_code == 400
+@pytest.mark.parametrize('url,status_code', [
+    ('/?a=1&b=2', 200),
+    ('/?a=wrong&b=parameters', 400),
+    ('/', 400),
+])
+def test_successful_api_call(client, url, status_code):
+    response = client.get(url)
+    assert response.status_code == status_code
 
 
 def test_only_get_is_allowed(client):
